server/src/test/java/io/crate/expression/tablefunctions/AbstractTableFunctionsTest.java

```diff
@@ -58,16 +61,31 @@ public void prepareFunctions() throws Exception {
         Symbol functionSymbol = sqlExpressions.normalize(sqlExpressions.asSymbol(expr));
 
         var function = (Function) functionSymbol;
-        TableFunctionImplementation<?> functionImplementation = (TableFunctionImplementation<?>) functions.getQualified(
+        var functionImplementation = (TableFunctionImplementation<?>) functions.getQualified(
             function,
             txnCtx.sessionSettings().searchPath()
         );
+
+        var boundArguments = functionImplementation.boundSignature().getArgumentTypes();
+        if (boundArguments.size() > 1 && boundArguments.stream()
+            .allMatch(arg -> arg.getBaseTypeName().equalsIgnoreCase(ArrayType.NAME))) {
+
+            var boundReturnType = functionImplementation.boundSignature().getReturnType();
+            assertThat(
+                "If the function has multiple fields, than the bound type must be RowType",
+                boundReturnType.getBaseTypeName(),
+                is(RowType.NAME)
+            );
+            assertThat(boundReturnType.getParameters().size(), greaterThan(1));
+        }
```

Comment:
I was thinking of something like this:


```suggestion
        if (functionImplementation.returnType().numElements() > 1) {
            // See classdocs of TableFunctionImplementation for an explanation
            assertThat(
                "If the rowType has multiple elements, the returnType of the boundSignature must be an exact match of the returnType",
                functionImplementation.boundSignature().getReturnType().createType(),
                is(functionImplementation.returnType())
            );
        }
```

What do you think?

Reply:
ah sure, we can assert against the `TableFunctionImplementation#returnType` :( it is defientely better like that!